libraries/abstractions/common_io/test/test_iot_flash.c

```diff
@@ -1597,146 +1180,89 @@ TEST( TEST_IOT_FLASH, AFQP_IotFlashWriteProtectWriteFailure )
+        if( pxFlashInfo->ucAsyncSupported )
+        {
+            /* Wait for the Erase to be completed and callback is called. */
+            lRetVal = xSemaphoreTake( xtestIotFlashSemaphore, portMAX_DELAY );
+            TEST_ASSERT_EQUAL( pdTRUE, lRetVal );
+        }
 
-    /* Check if block protection is supported */
-    if( lRetVal != IOT_FLASH_FUNCTION_NOT_SUPPORTED )
-    {
-        TEST_ASSERT_EQUAL( IOT_FLASH_SUCCESS, lRetVal );
+        /* Make the sector as read only. */
+        xFlashProtect.ulAddress = ultestIotFlashStartOffset;
+        xFlashProtect.ulSize = pxFlashInfo->ulLockSupportSize;
+        xFlashProtect.xProtectionLevel = eFlashReadOnly;
 
-        /* Read back the block protection */
         lRetVal = iot_flash_ioctl( xFlashHandle,
-                                   eGetFlashBlockProtection,
+                                   eSetFlashBlockProtection,
                                    ( void * const ) &xFlashProtect );
 
-        TEST_ASSERT_EQUAL( IOT_FLASH_SUCCESS, lRetVal );
-
-        /* Fill out a buffer of sectorSize to be written */
-        for( uint32_t i = 0; i < pxFlashInfo->ulPageSize; i++ )
+        /* Check if block protection is supported */
+        if( lRetVal != IOT_FLASH_FUNCTION_NOT_SUPPORTED )
         {
-            uctestIotFlashWriteBuffer[ i ] = i;
-        }
+            TEST_ASSERT_EQUAL( IOT_FLASH_SUCCESS, lRetVal );
 
-        /* Try writing to this sector */
-        lRetVal = iot_flash_write_sync( xFlashHandle,
-                                        ultestIotFlashStartOffset,
-                                        &uctestIotFlashWriteBuffer[ 0 ],
-                                        pxFlashInfo->ulPageSize );
-        TEST_ASSERT_EQUAL( IOT_FLASH_SUCCESS, lRetVal );
+            /* Read back the block protection */
+            lRetVal = iot_flash_ioctl( xFlashHandle,
+                                       eGetFlashBlockProtection,
+                                       ( void * const ) &xFlashProtect );
 
-        /* Read back the data written */
-        lRetVal = iot_flash_read_sync( xFlashHandle,
-                                       ultestIotFlashStartOffset,
-                                       &uctestIotFlashReadBuffer[ 0 ],
-                                       pxFlashInfo->ulPageSize );
-        TEST_ASSERT_EQUAL( IOT_FLASH_SUCCESS, lRetVal );
+            TEST_ASSERT_EQUAL( IOT_FLASH_SUCCESS, lRetVal );
 
-        /* Verify that the data is not written to the flash */
-        for( uint32_t i = 0; i < pxFlashInfo->ulPageSize; i++ )
-        {
-            if( uctestIotFlashReadBuffer[ i ] != testIotFLASH_DEFAULT_FLASH_BYTE )
-            {
-                TEST_ASSERT_MESSAGE( 0, "Contents do NOT match when readback" );
-            }
+            /* Write the a full page and verify the data by reading it back */
+            prvIotFlashWriteReadVerify( xFlashHandle, ultestIotFlashStartOffset, pxFlashInfo->ulPageSize );
```

Comment:
This function writes and reads the same data. When reading back - that data should be erased data, and not written data. Please modify this call to write and then ReadVerifyErased

Reply:
Good catch, fixing this asap.